### PR TITLE
Add prototype of shell support

### PR DESCRIPTION
**Pull request by joshma**
This doesn't have everything yet (primarily missing params), but this allows you to
deploy a `script.sh` file and, optionally, looks for the nearest `Dockerfile` to build
the surrounding environment. (If not found, it uses a hardcoded `ubuntu` image.)

### CODE REVIEW

File: pkg/build/builder.go [hunk 1]
```diff
@@ -148,7 +148,8 @@ func (b *Builder) Build(ctx context.Context, taskID, version string) (*Response,
 	}
 	logger.Debug(strings.TrimSpace(dockerfile))
 
-	if err := tree.Write("Dockerfile", strings.NewReader(dockerfile)); err != nil {
+	dockerfilePath := "Dockerfile.airplane"
```
Comment (joshma) [inline pkg/build/builder.go:151]: Without this, we were attempting to override an existing `Dockerfile` (i.e. the one being used for the script!) and it wasn't actually successfully writing.

File: pkg/build/shell-shim.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# TODO: set up env vars - for each argument, split by first =, set AP_FIRST_PART=SECOND_PART
```
Comment (joshma) [inline pkg/build/shell-shim.sh:3]: Doing this in a separate PR

File: pkg/build/shell.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+	// Extend template with our own logic - set up a WORKDIR and shim.
+	dockerfileTemplate = dockerfileTemplate + heredoc.Doc(`
+		WORKDIR /airplane
+		RUN mkdir -p .airplane && {{.InlineShim}} > .airplane/shim.sh
+		
+		COPY . .
+		RUN chmod +x {{.Entrypoint}}
+		
+		ENTRYPOINT ["bash", ".airplane/shim.sh"]
```
Comment (joshma) [inline pkg/build/shell.go:83]: All this does right now is `exec` into the entrypoint, but in the future it'll set up env vars from params and possibly other things.

File: pkg/build/shell.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/airplanedev/cli/pkg/api"
+	"github.com/airplanedev/cli/pkg/fsx"
+	"github.com/pkg/errors"
+)
+
+func shell(root string, options api.KindOptions) (string, error) {
+	// Assert that the entrypoint file exists:
+	entrypoint, _ := options["entrypoint"].(string)
```
Comment (francescaleung) [inline pkg/build/shell.go:16]: Not a bad idea to check the second return value here

File: pkg/runtime/shell/shell.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	"github.com/airplanedev/cli/pkg/api"
+	"github.com/airplanedev/cli/pkg/fsx"
+	"github.com/airplanedev/cli/pkg/logger"
+	"github.com/airplanedev/cli/pkg/runtime"
+	"github.com/pkg/errors"
+)
+
+// Init register the runtime.
+func init() {
+	runtime.Register(".sh", Runtime{})
```
Comment (francescaleung) [inline pkg/runtime/shell/shell.go:20]: This means you can't deploy a script without a file extension, right?

**Review by francescaleung — APPROVED** [review]

File: pkg/build/shell.go (hunk 4 above)
Comment (joshma) [reply pkg/build/shell.go:16]: Can you explain? This'll be empty string if it's !ok, so this handles both cases in one.

File: pkg/runtime/shell/shell.go (hunk 5 above)
Comment (joshma) [reply pkg/runtime/shell/shell.go:20]: Yeah, correct - for now I'm thinking we start with this, and require .sh. I was thinking to generalize we'd bring back `.yml` support and you'd deploy the `yml` file with an explicit `entrypoint` set to the possibly-extensionless file

**Comment by joshma** [comment]
cc @colinking - you should do a post-merge review on this

File: pkg/build/shell.go (hunk 4 above)
Comment (joshma) [reply pkg/build/shell.go:16]: I guess I can handle empty string more elegantly, right now it errosr below

File: pkg/build/builder.go (hunk 1 above)
Comment (colinking) [reply pkg/build/builder.go:151]: Could we do `.airplane/Dockerfile` instead?

Comment (joshma) [reply pkg/build/builder.go:151]: Yeah good idea - https://github.com/airplanedev/cli/pull/228

**Pull request merged**